torch/_dynamo/variables/builtin.py

```diff
@@ -1620,17 +1620,27 @@ def _unimplemented():
         if isinstance(left, TensorVariable) or isinstance(right, TensorVariable):
             from .builder import wrap_fx_proxy_cls
 
-            if op in [operator.is_, operator.is_not]:
-                is_result = (
-                    isinstance(left, TensorVariable)
-                    and isinstance(right, TensorVariable)
-                    and id(extract_fake_example_value(left.as_proxy().node))
-                    == id(extract_fake_example_value(right.as_proxy().node))
-                )
-                if op is operator.is_:
-                    return ConstantVariable.create(is_result)
-                else:
-                    return ConstantVariable.create(not is_result)
+            if op is operator.is_ or op is operator.is_not:
+
+                def return_constant(bool_value):
+                    if op is operator.is_:
+                        return ConstantVariable.create(bool_value)
+                    else:
+                        return ConstantVariable.create(not bool_value)
+
+                if isinstance(left, TensorVariable) != isinstance(
+                    right, TensorVariable
+                ):
+                    return return_constant(False)
+                elif id(extract_fake_example_value(left.as_proxy().node)) == id(
+                    extract_fake_example_value(right.as_proxy().node)
+                ):
+                    # if the above condition is false, result actual result can still be
+                    # true ex:
+                    #  x = np.True_
+                    #  z = x | x
+                    #  assert(x is z) => should be true, would give false
+                    return return_constant(True)
```

Comment:
Do we wrap ```np.True_``` as ```NumpyNdarrayVariable```? I don't think we should do that. If we wrap ```np.True_``` as ```ConstantVariable```, then z is another constant, then ```x is z``` should not fall into this bucket.

Reply:
but then for Z in that case to be constant we actually are depending on the const folding logic of the or, to be applied , but can we gurantee that for all possible functions

Reply:
I see even if we do not do const folding in that case, it would be ConstantVariable compared with NumpyNdarrayVariable and we wont have issue also.  

@Skylion007 what is your thoughts here?
I wonder if its possible to define two functions f1(), f2() such that f1() and f2() outputs are the same (is wise)
but they appear not the same as different NumpyNdarrayVariable